src/lib.rs

```diff
@@ -105,51 +105,69 @@ pub mod sign;
 pub mod transaction;
 pub mod transport;
 
-use crate::contract::{
-    DeployBuilder, DeployedFuture, EventBuilder, EventStream, MethodBuilder, ViewMethodBuilder,
-};
-pub use crate::contract::{Event, Instance, Topic};
-pub use crate::int::I256;
-use crate::log::LogStream;
-pub use crate::transaction::{Account, GasPrice};
-pub use crate::transport::DynTransport;
+pub use crate::contract::Instance;
+pub use crate::prelude::*;
 pub use ethcontract_common as common;
 pub use ethcontract_common::truffle::Artifact;
 #[cfg(feature = "derive")]
 pub use ethcontract_derive::contract;
 pub use jsonrpc_core as jsonrpc;
-pub use secret::{Password, PrivateKey};
 pub use serde_json as json;
 pub use web3;
-use web3::api::Web3;
-pub use web3::types::{Address, BlockNumber, TransactionCondition, H160, H256, U256};
 
-/// Type alias for a `Web3` provider with an underlying `DynTransport`.
-pub type DynWeb3 = Web3<DynTransport>;
+pub mod prelude {
+    //! A prelude module for importing commonly used types when interacting with
+    //! generated contracts.
+
+    pub use crate::contract::{Event, Topic};
+    pub use crate::int::I256;
+    pub use crate::secret::{Password, PrivateKey};
+    pub use crate::transaction::{Account, GasPrice};
+    pub use web3::api::Web3;
+    pub use web3::transports::Http;
+    pub use web3::types::{Address, BlockNumber, TransactionCondition, H160, H256, U256};
+}
+
+pub mod dyns {
+    //! Type aliases to various runtime types that use an underlying
+    //! `DynTransport`. These types are used extensively throughout the
+    //! generated code.
+
+    use crate::contract::{
+        DeployBuilder, DeployedFuture, EventBuilder, EventStream, Instance, MethodBuilder,
+        ViewMethodBuilder,
+    };
+    use crate::log::LogStream;
+    pub use crate::transport::DynTransport;
+    use web3::api::Web3;
```

Comment:
Do we need to reimport Web3 even though the super module already has it?

Reply:
Yes, we would have to use `use super::Web3`